analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -337,4 +346,95 @@ private void writeRepositorySettingsFile(Collection<ArtifactRepository> reposito
             getProject().getLogger().info("Repository export disabled.");
         }
     }
+
+    //TODO this needs to be improved since it is very naive
+    private void findDynamicDependencies(org.gradle.api.artifacts.Configuration configuration,
+            Set<ProjectRef> dynamicDependencies, Map<ProjectRef, String> dynamicDepToKey) {
+        configuration.getIncoming().getDependencies().withType(ExternalDependency.class).matching(d -> {
+            final String requiredVersion = d.getVersionConstraint().getRequiredVersion();
+            return requiredVersion.endsWith("latest.release") || requiredVersion.endsWith("+");
+        }).forEach(d -> {
+            final SimpleProjectRef ref = new SimpleProjectRef(d.getGroup(), d.getName());
+            dynamicDependencies.add(ref);
+            dynamicDepToKey.put(ref,
+                    new SimpleProjectVersionRef(ref, d.getVersionConstraint().getRequiredVersion()).toString());
+        });
```

Comment:
I think we can use the DefaultVersionSelector which should give a much better determination of whether something is dynamic or not. Further I think we need to maintain a mapping between original dep (e.g. 1.1+) and what is passed to DA

Reply:
Now that the tests are in place, I can add a wrapper around ProjectVersionRef that includes the original version

Reply:
See my above comment - I want to see whether its feasible to take another look and see if its possible to make things simpler rather than more wrappers around things.

Reply:
One point regarding this, while its cleaner (!) than my pseudo code earlier I notice its before we handle unresolved dependencies - I wonder if that could be an issue?

Reply:
The actual resolution goes on later on, this part should just be book keeping for later on.
Of course there might be something I have missed, I wouldn't be surprised when it comes to Gradle 😆